zipline/utils/events.py

```diff
@@ -539,7 +539,7 @@ def get_first_trading_day_of_month(self, dt):
         self.month = dt.month
 
         dt = dt.replace(day=1)
-        first_day = (dt if self.cal.is_open_on_day(dt)
+        first_day = (normalize_date(dt) if self.cal.is_open_on_day(dt)
```

Comment:
Yup, I saw this.  Leave it as `normalize_date` now, I don't trust the existing `session_date` method to do the right thing.  I'll go through the usages of `normalize_date` later and sort it out.